snarkos/network/mod.rs

```diff
@@ -119,7 +119,7 @@ pub(crate) async fn handle_peer<N: Network>(
                             // Check if the block can be added to the ledger.
                             if block.height() == ledger.ledger().read().latest_height() + 1 {
```

Comment:
this check is not needed, as it happens in `add_next_block` (when `check_next_block` is called: https://github.com/AleoHQ/snarkVM/blob/testnet3/vm/compiler/src/ledger/mod.rs#L350)

Reply:
Agreed, this is a redundant check. Initially added to clean up logging, since we didn't want to output a "Failed to process block..." warning for every misaligned block and instead declare we are "skipping" it.

Reply:
if we want to keep it, it'd probably be best to put it in the [`add_next_block`](https://github.com/AleoHQ/snarkOS/pull/1880/files#diff-a965d43412c665e2554b72a2041a4491c9267d2280f9688d349f282487e25b3bR123) method introduced by this PR

Reply:
though this would complicate the signature a bit; let's skip this for now